virttest/utils_test/libvirt.py

```diff
@@ -3760,3 +3761,354 @@ def check_dumpxml(vm, content, err_ignore=False):
+                                          "global_quota": "cpu.max",
+                                          "iothread_period": "<iothreadX>/cpu.max",
+                                          "iothread_quota": "<iothreadX>/cpu.max"}
+
+    def __init__(self, vm_pid):
+        self.__vm_pid = vm_pid
+
+    def is_cgroup_v2_enabled(self):
+        """
+        Check if cgroup v2 enabled on host
+
+        :return: True means cgroup v2 enabled
+                 False means not
+        """
+        cmd = "cat /proc/mounts | grep cgroup2"
+        cmd_result = process.run(cmd, ignore_status=True, shell=True)
+        if cmd_result.exit_status:
+            return False
+        return True
+
+    def get_cgroup_path(self, controller=None):
+        """
+        Get specific cgroup controller's root path
+
+        :params controller: The cgroup controller, used for cgroup v1. For
+                            cgroup v2 this param will be ignored since all
+                            controllers are in the same dir
+        :return: The path to the cgroup controller
+        """
+        cgroup_path = ""
+        if self.is_cgroup_v2_enabled():
+            vm_proc_cgroup_path = "/proc/%s/cgroup" % self.__vm_pid
+            with open('/proc/mounts', 'r') as mnts:
+                cg_mount_point = re.findall(r"\s(\S*cgroup)\s", mnts.read())
+            with open(vm_proc_cgroup_path, 'r') as vm_cg_file:
+                cg_vm_scope = re.findall(r"\S*::(\S*)", vm_cg_file.read())
+            cgroup_path = os.path.join(cg_mount_point[0],
+                                       cg_vm_scope[0].strip("/"))
+            if "emulator" in cgroup_path:
+                cgroup_path += "/.."
+        else:
+            cgroup_path = utils_cgroup.resolve_task_cgroup_path(
+                    int(self.__vm_pid), controller)
+        if not os.path.exists(cgroup_path):
+            logging.error("cgroup path '%s' doesn't exist" % cgroup_path)
+            return None
+        return cgroup_path
+
+    def __get_cpu_subdirs(self, controller_path=None, dir_keyword=None):
+        """
+        Search and return the sub dirs of the cpu controllers by keyword
+
+        :param controller_path: The path of the cpu controller
+        :param dir_keyword: The keyword of the sub dirs. Normally it could be
+                            "vcpu", "emulator", "iothread"
+        :return: The list of the sub dir names
+        """
+        dir_names = []
+        for filename in os.listdir(controller_path):
+            if dir_keyword in filename:
+                dir_names.append(filename)
+        if not dir_names and "iothread" in dir_keyword:
+            logging.debug("No sub dirs found with keyword: '%s'. "
+                          "Pls check if you've executed virsh cmd "
+                          "'iothreadadd'.", dir_keyword)
+            return None
+        return sorted(dir_names)
+
+    def __get_standardized_cgroup1_info(self, virsh_cmd=None):
```

Comment:
maybe enforce virsh_cmd must be given one value, rather than None as default value

Reply:
No suitable virsh cmd here is a suitable 'default' command. And None will trgger a error notice later in these methods